typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl

```diff
@@ -243,3 +243,98 @@ anticipate with their pattern matching, e.g.:
 
   (eval:error
    (size 42))]
+
+
+@section{@racket[is-a?] and Occurrence Typing}
+
+Typed Racket does not use the @racket[is-a?] predicate to refine object types
+ because the target object may have been created in untyped code and
+ @racket[is-a?] does not check the types of fields and methods.
+
+For example, the code below defines a class type @racket[Pizza%], a subclass
+ type @racket[Sauce-Pizza%], and a function @racket[get-sauce] (this
+ function contains a type error).
+The @racket[get-sauce] function uses @racket[is-a?] to test the class of its
+ argument; if the test is successful, the function expects the argument to have
+ a field named @racket[topping] that contains a value of type @racket[Sauce].
+
+@codeblock{
```

Comment:
I wanted this to be an `@examples`, but I couldn't get the error message to point to the right expression. I'd see this:

```
eval:33:0: Type Checker: type mismatch
expected: Sauce
given: Any
in: bbq
```

instead of `in: (get-field topping pizza)`